ikarus/finiteelements/mechanics/enhancedassumedstrains.hh

```diff
@@ -170,30 +154,24 @@ forward the
 
     auto resultVector = DisplacementBasedElement::template calculateAt<resType>(req, local);
 
-    if (isDisplacementBased())
-      return resultVector;
-
     const auto C              = DisplacementBasedElement::materialTangentFunction(req);
     const auto& numberOfNodes = DisplacementBasedElement::numberOfNodes();
     auto uFunction            = DisplacementBasedElement::displacementFunction(req);
-    const auto disp           = Dune::viewAsFlatEigenVector(uFunction.coefficientsRef());
-
-    std::visit(
-        [&]<typename EAST>(const EAST& easFunction) {
-          if constexpr (not std::is_same_v<std::monostate, EAST>) {
-            constexpr int enhancedStrainSize = EAST::enhancedStrainSize;
-            Eigen::Matrix<double, enhancedStrainSize, enhancedStrainSize> D;
-            calculateDAndLMatrix(easFunction, req, D, L_);
-            const auto alpha = (-D.inverse() * L_ * disp).eval();
-            const auto M     = easFunction.calcM(local);
-            const auto CEval = C(local);
-            auto easStress   = (CEval * M * alpha).eval();
-
-            resultVector.resize(3, 1);
-            resultVector = resultVector + easStress;
-          }
-        },
-        easVariant_);
+
+    auto calculateAtContribution = [&]<typename EAST>(const EAST& easFunction) {
+      typename EAST::DType D;
+      calculateDAndLMatrix(easFunction, req, D, L_);
+
+      const auto disp  = Dune::viewAsFlatEigenVector(uFunction.coefficientsRef());
+      const auto alpha = (-D.inverse() * L_ * disp).eval();
+      const auto M     = easFunction.calcM(local);
+      const auto CEval = C(local);
+      auto easStress   = (CEval * M * alpha).eval();
+
+      resultVector.resize(3, 1);
```

Comment:
Not your fault but this resize seems strange here, since `resultVector` is alread populated. This is not needed and the line below can also be +=.

Reply:
This should already be fixed in #261